source/local/source.go

```diff
@@ -123,11 +137,20 @@ func (ls *localSourceHandler) CacheKey(ctx context.Context, g session.Group, ind
 		sessionID = id
 	}
 	dt, err := json.Marshal(struct {
-		SessionID       string
-		IncludePatterns []string
-		ExcludePatterns []string
-		FollowPaths     []string
-	}{SessionID: sessionID, IncludePatterns: ls.src.IncludePatterns, ExcludePatterns: ls.src.ExcludePatterns, FollowPaths: ls.src.FollowPaths})
+		SessionID          string
+		IncludePatterns    []string
+		ExcludePatterns    []string
+		FollowPaths        []string
+		MetadataTransfer   bool     `json:"metadataTransfer,omitempty"`
+		MetadataExceptions []string `json:"metadataExceptions,omitempty"`
```

Comment:
I think we should keep the same casing or update the others:

```suggestion
		MetadataTransfer   bool     `json:",omitempty"`
		MetadataExceptions []string `json:",omitempty"`
```